src/include/txn.i

```diff
@@ -831,19 +838,24 @@ __wt_upd_alloc_tombstone(WT_SESSION_IMPL *session, WT_UPDATE **updp, size_t *siz
  *     Get the first visible update in a list (or NULL if none are visible).
  */
 static inline int
-__wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd)
+__wt_txn_read_upd_list(
+  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT_UPDATE **prepare_updp)
 {
     WT_VISIBLE_TYPE upd_visible;
     uint8_t type;
 
+    if (prepare_updp != NULL)
+        *prepare_updp = NULL;
     __wt_upd_value_clear(cbt->upd_value);
 
     for (; upd != NULL; upd = upd->next) {
         WT_ORDERED_READ(type, upd->type);
         /* Skip reserved place-holders, they're never visible. */
         if (type == WT_UPDATE_RESERVE)
             continue;
-        upd_visible = __wt_txn_upd_visible_type(session, upd);
+        /* Save the first prepared update we see. */
+        upd_visible = __wt_txn_upd_visible_type(
+          session, upd, prepare_updp != NULL && *prepare_updp == NULL ? prepare_updp : NULL);
```

Comment:
Couldn't this code do 
```
if (upd_visible == WT_VISIBLE_FALSE && F_ISSET(session->txn, WT_TXN_IGNORE_PREPARE) { 
    prepare_upd = upd;
}
```

To save pushing this new parameter down into `__wt_txn_upd_visible_type`. I'd really like to avoid adding the prepare parameter to `__wt_txn_read_upd_list` as well. It's going to be hard to remember that semantic. Could  __wt_txn_read be restructured to save doing the work for all callers here?

Reply:
I also don't like this but I don't see how that can be done ATM. Does anyone have a better idea?

Reply:
How about doing the  `ignore_prepare` check in the caller function and decide like continuing on `ignore_prepare` or return prepare conflict?

Reply:
OK, I'll try. But we still need to do that in `__wt_txn_read_upd_list ` I think.